Subject: [PATCH] Do not pass --disable-new-dtags to the linker

Android's dynamic linker only honours RUNPATH (DT_RUNPATH), not the
legacy DT_RPATH that --disable-new-dtags would emit.

Rebased for jdk26: the linker flag setup was moved from
make/autoconf/flags-cflags.m4 to make/common/JdkNativeCompilation.gmk.

--- a/make/common/JdkNativeCompilation.gmk
+++ b/make/common/JdkNativeCompilation.gmk
@@ -35,9 +35,9 @@
   # This protects internal library dependencies within the JDK from being
   # overridden using LD_LIBRARY_PATH. See JDK-8326891 for more information.
   SetExecutableOrigin = \
-      -Wl,-rpath,\$(DOLLAR)ORIGIN$1 -Wl,--disable-new-dtags
+      -Wl,-rpath,\$(DOLLAR)ORIGIN$1
   SetSharedLibraryOrigin = \
-      -Wl,-z,origin -Wl,-rpath,\$(DOLLAR)ORIGIN$1 -Wl,--disable-new-dtags
+      -Wl,-z,origin -Wl,-rpath,\$(DOLLAR)ORIGIN$1
 else ifeq ($(call isCompiler, clang), true)
   ifeq ($(call isTargetOs, macosx), true)
     SetExecutableOrigin = \
@@ -50,7 +50,7 @@
   else
     ifeq ($(call isTargetOs, linux), true)
       SetExecutableOrigin = \
-          -Wl,-rpath,\$(DOLLAR)ORIGIN$1 -Wl,--disable-new-dtags
+          -Wl,-rpath,\$(DOLLAR)ORIGIN$1
     else
       SetExecutableOrigin = \
           -Wl,-rpath,\$(DOLLAR)ORIGIN$1
